### ISSUE TITLE
adjustment of content to window size

### ISSUE THREAD

**Issue by pmattern** [hagabaka/sddm-config-editor#1]
Content doesn't get adapted to changing window size which can result in garbled output.

The main window doesn't seem to be able to wrap the actual content or adjust it to decreasing window size by any other means. As a consequence content starts overlapping if width and/or height of a window decrease.
![sddm-config-editor_overlapping-content](https://cloud.githubusercontent.com/assets/9860801/8476446/b9c2e41c-20c1-11e5-83c7-986ad5b47946.png)

Sometimes the headers get cut besides sufficient horizontal space is available.
![sddm-config-editor_cropped-headers](https://cloud.githubusercontent.com/assets/9860801/8476449/c3455b64-20c1-11e5-9df2-aecdb89fed78.png)

**Comment by hagabaka** [comment]
The window is already tightly packed, so adjusting layout on resize probably won't help much.

I think the best solution is to display only one configuration section at a time, and have a tab bar or list box to switch between sections. That should also solve your problem with section headers coincidentally (although I haven't been able to reproduce it). What do you think?

**Comment by pmattern** [comment]
Actually I still think it could be an option to just wrap the text and make those forms to fill in paths and the like adjust their size automatically if this is technically feasible.

Maybe revamping the UI in general is the better option, though. If you should consider this I for one would find an UI as used to configure LXQt's panel
![lxqt_panel_dialog-configure-panel](https://cloud.githubusercontent.com/assets/9860801/8503675/2805f30e-21c3-11e5-82a9-1fe91e73e43c.png)
the most convenient.

The cutting of headers could be seen on Xfce only, no matter whether Openbox or Xfwm were running, neither on KDE Plasma 5 nor LXQt.
Maybe this is because on Xfce only those headers are in bold font (for whatever reason, didn't try to look into this any further).

**Comment by hagabaka** [comment]
In a06bf050865d82ab38a2e136ab2a3c756acb5c9e I switched to a tab based UI. It still can't make adjustments on window resize, but the window should fit in smaller screens easily now.

It should be easy to convert the tabs to a vertical list like configuration dialogs in LXQt and KDE, but before working on that, I wanted to see if this version solves any of your problems.

**Comment by pmattern** [comment]
Just had a look at a06bf05:
Still it's not possible to wrap text which seems a bit odd to me. But probably this won't matter any longer due to the new layout.
Those cut headers are gone as well, both on LXQt/Openbox and Xfce/Xfwm. If horizontal space decreases characters are replaced by '...' which makes senses, IMO.

Notes (unrelated stuff I deem worth mentioning but not worth putting into an issue on its own or so):
Depending on the font explanatory text may be bold and larger than the text it refers to like here (DejaVu Sans 10)
![sddm-config-editor_bold-explanatory-text](https://cloud.githubusercontent.com/assets/9860801/8608866/97aaf138-269f-11e5-8a47-b29b24994b71.png)
When using those tabs or something similar I'm not sure whether alphabetical order of headers is intuitive. Not sure how to do it better either, though.
Aside from this I wonder whether it wouldn't make sense to integrate 'Autologin' into 'Users'.
sddm.conf is saved 600. Isn't this a bit strict? IMHO 644 would do as well as there are no credentials in that file and allow ordinary users to quickly check which options are set.
Latest commit / afd6003 doesn't build here but fails with some error messages regarding translation I can't judge.

**Comment by hagabaka** [comment]
I've fixed the build error about translations. Note that you need to have `lrelease` (part of `qt5-tools` on ArchLinux).

To minimize hard-coding, sddm-config-editor displays settings in the same structure as the sddm configuration file, so merging Autologin and Users sections isn't an option.

The program doesn't change the mode of sddm.conf, but I guess it uses mode 600 to create it if sddm.conf doesn't exist before running it. This shouldn't happen often, but I will fix it later.

**Comment by pmattern** [comment]
> The program doesn't change the mode of sddm.conf, but I guess it uses mode 600 to create it if sddm.conf doesn't exist before running it. This shouldn't happen often, but I will fix it later.

You're right, this is exactly what happens: if sddm.conf doesn't exist yet, it's saved 600, if it exists, the permission bits aren't touched.
Not sure whether a missing sddm.conf is that seldom, btw. E. g. Arch Linux doesn't ship it by default.
